compiler/src/iree/compiler/Codegen/TransformDialectStrategies/GPU/StagedReductionStrategy.cpp

```diff
@@ -112,9 +113,20 @@ void mlir::iree_compiler::gpu::StagedReductionStrategy::configure(
   }
 }
 
+static Value shareForeachArgument(ImplicitLocOpBuilder &b, Value foreachThread,
+                                  ArrayRef<int64_t> indices) {
+  auto foreachType =
+      transform::OperationType::get(b.getContext(), "scf.foreach_thread");
```

Comment:
Nit: `scf::ForeachThreadOp::getOperationName()` to avoid magic strings. I'll add a convenience getter upstream for this.